com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/validation/Validator.java

```diff
@@ -266,6 +269,31 @@ protected void checkMissingRequiredProperties(Set<SwaggerError> errors, Abstract
         }
     }
 
+    /**
+     * This method checks that referenced objects are of expected type as defined in the schema.
+     * 
+     * @param errors
+     * @param node
+     */
+    protected void checkReferenceType(Set<SwaggerError> errors, AbstractNode node) {
+        if (JsonReference.isReference(node)) {
+            Model model = node.getModel();
+            TypeDefinition type = node.getType();
+
+            if (type instanceof ReferenceTypeDefinition) {
+                type = ((ReferenceTypeDefinition) type).resolve();
+            }
+
+            AbstractNode nodeValue = node.get(JsonReference.PROPERTY);
+            AbstractNode valueNode = model.find((String) nodeValue.asValue().getValue());
+
+            if (valueNode != null && type.getPointer() != null && valueNode.getType() != null
+                    && !type.getPointer().equals(valueNode.getType().getPointer())) {
+                errors.add(error(nodeValue, IMarker.SEVERITY_ERROR, Messages.error_invalid_reference_type));
```

Comment:
It would be **really** helpful to provide what type is actually expected. It looks like `valueNode.getType().getPointer()` can provide this info